models/maespa/R/met2model.MAESPA.R

```diff
@@ -0,0 +1,265 @@
+        hr <- c(hr,rep(NA,length(dtmp)))
+      }
+      rng <- length(doy) - length(ytmp):1 + 1
+      if(!all(rng>=0)){
+        skip = TRUE
+        logger.warn(paste(year,"is not a complete year and will not be included"))
+        break
+      }
+      asec[rng] <- asec[rng] - asec[rng[1]]
+      hr[rng] <- (asec[rng] - (dtmp-1)*86400)/86400*24
+    }
+    if(length(yr) < length(sec)){
+      rng <- (length(yr)+1):length(sec)
+      if(!all(rng>=0)){
+        skip = TRUE
+        logger.warn(paste(year,"is not a complete year and will not be included"))
+        break
+      }
+      yr[rng] <- rep(y+1,length(rng))
+      doy[rng] <- rep(1:366,each=86400/dt)[1:length(rng)]
+      hr[rng] <- rep(seq(0,length=86400/dt,by=dt/86400*24),366)[1:length(rng)]
+    }
+    if(skip){
+      print("Skipping to next year")
+      next
+    }
+    tmp<-rbind(TAIR,PPT)
+    
+    if(is.null(out)){
+      out = tmp
+    } else {
+      out = cbind(out,tmp)}
+    
+  }### end loop over years
+  
+  
+  DOY = doy
+  #Bind met data and get rid of empty parmeters
+  
+  out[is.na(out)] <-0
+  #Get names
+  columnnames =  paste0("'",rownames(out),"'",collapse= " ")
+  #Get number of variables
+  numbercolumns = nrow(out)
+  #turn into matrix
+  out<- matrix(out,ncol= numbercolumns)
+  
+  #Set day or hour Option(1 or 0)
+  if(tstep>=1){dayorhour=1}else{dayorhour=0}
+  #Set number of timesteps in a day(timetsep of input data)
+  timesteps = tstep
+  # Set distribution of diffuse radiation incident from the sky.(0.0) is default. 
+  difsky= 0.5
+  #Change format of date to DD/MM/YY
+  startdate = paste0("'",format(as.Date(start_date),"%d/%m/%y"),"'")
+  enddate = paste0("'",format(as.Date(end_date),"%d/%m/%y"),"'")
+  metdat <- readLines(con=system.file("template.met", package = "PEcAn.MAESPA"), n=-1)
+  
+  
+  ## write output
+  #metdat<- gsub('@MAESPAMETHEADER@',metheader,metdat)
+  metdat<- gsub('@DISTDIFFRADINCIDENCE@',difsky,metdat)
+  metdat<- gsub('@DEFCO2@',defaultCO2,metdat)
+  #metdat<- gsub('@DEFSWMIN@',defaultSWmin,metdat)
+  #metdat<- gsub('@DEFSWMAX@',defaultSWmax,metdat)
+  #metdat<- gsub('@DEFATMOSPRESSURE@',deafaultatmospress,metdat)
+  metdat<- gsub('@LATITUDE@',lat,metdat)
+  metdat<- gsub('@LONGITUDE@',lon,metdat)
+  #metdat<- gsub('@TZLONG@',longmeridian,metdat)
+  #metdat<- gsub('@LONGHEM@',longhemisphere,metdat)
+  #metdat<- gsub('@LATHEM@',lathemisphere,metdat)
+  metdat<- gsub('@DAYORHR@',dayorhour,metdat)
+  metdat<- gsub('@TSTEPS@',timesteps,metdat)
+  metdat<- gsub('@NUMCOLUMNS@',numbercolumns,metdat)
+  metdat<- gsub('@STARTDATE@',startdate,metdat)
+  metdat<- gsub('@ENDDATE@',enddate,metdat)
+  metdat<- gsub('@COLNAMES@',columnnames,metdat)
+  
+  writeLines(metdat, con=file.path(outfolder, "met.dat"))
+  write(paste(out),file=file.path(outfolder, "met.dat"),append = TRUE,ncol=numbercolumns)
+  writeLines(metdat, con=file.path(out.file.full))
+  write(paste(out),file=file.path(out.file.full),append = TRUE,ncol=numbercolumns)
```

Comment:
The 4 lines above appear to write the same data to two different files -- is there a reason for that?